pymeasure/instruments/instrument.py

```diff
@@ -93,36 +93,11 @@ def __set_name__(self, owner, name):
         self.name = name
 
 
-class Instrument:
-    """ The base class for all Instrument definitions.
-
-    It makes use of one of the :py:class:`~pymeasure.adapters.Adapter` classes for communication
-    with the connected hardware device. This decouples the instrument/command definition from the
-    specific communication interface used.
-
-    When ``adapter`` is a string, this is taken as an appropriate resource name. Depending on your
-    installed VISA library, this can be something simple like ``COM1`` or ``ASRL2``, or a more
-    complicated
-    `VISA resource name <https://pyvisa.readthedocs.io/en/latest/introduction/names.html>`__
-    defining the target of your connection.
-
-    When ``adapter`` is an integer, a GPIB resource name is created based on that.
-    In either case a :py:class:`~pymeasure.adapters.VISAAdapter` is constructed based on that
-    resource name.
-    Keyword arguments can be used to further configure the connection.
-
-    Otherwise, the passed :py:class:`~pymeasure.adapters.Adapter` object is used and any keyword
-    arguments are discarded.
+class CommonBase:
```

Comment:
`instrument.py` is getting long. What do you think about splitting out `CommonBase` and `Channel` to `common.py` and `channel.py` modules, respectively? That would also make tracebacks like for the breaking doctest easier to parse, as you don't need to check line numbers to find out in which class you are.

Reply:
I see your point. They are, however, closely connected and sometimes it is better to stay in the same file to work with the `Instrument`, instead of jumpying between two files.

At the end, I think it is good to outsource dynamic property with CommonBase to their own file, as they contain all the pymeasure magic. The Instrument and Channel get their own files.